File: A19Sorting/A2SelectionSort.js
let arr = [5, 4, 1, 3, 2, 7];

function selectionSort(arr) {
  for (let turn = 0; turn < arr.length - 1; turn++) {
    let smallPos = turn;
    for (let j = turn; j < arr.length; j++) {
      if (arr[smallPos] > arr[j]) {
        smallPos = j;
      }
    }
    let temp = arr[smallPos];
    arr[smallPos] = arr[turn];
    arr[turn] = temp;
  }
  console.log("arr", arr);
}

selectionSort(arr);

File: A19Sorting/A2SelectionSort.java
package A19Sorting;

public class A2SelectionSort {
    // int arr[] = { 5, 4, 1, 3, 2, 7 };
    public static void SelectionSort(int arr[]) {
        for (int i = 0; i < arr.length - 1; i++) {
            int smallPos = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[smallPos] > arr[j]) {
                    smallPos = j;
                }
            }
            int temp = arr[smallPos];
            arr[smallPos] = arr[i];
            arr[i] = temp;
        }
        for (int j = 0; j < arr.length; j++) {
            System.out.println(arr[j]);
        }
    }

    public static void main(String[] args) {
        int arr[] = { 5, 4, 1, 3, 2, 7 };
        SelectionSort(arr);
    }
}
